react/overall-learning-objectives.md

```diff
@@ -0,0 +1,36 @@
+# Overall Learning Objectives - React module
+
+By the end of the three lessons & homework, students should be able to build a React application:
+
+- Written with components
+  - Appropriately composed/nested
+  - Split across multiple files
+- Uses props to pass values between components
+  - Data like numbers/strings/objects/arrays
+  - Functions as event handlers
+- Uses state to react to user interaction
+  - E.g. a button can be clicked to increment a counter
+- That fetches data from an API
+  - Loading, success and error states are shown
+- Containing a form
+  - Form inputs that can be filled out by the user
+  - Can be submitted, triggering an update in the app
+
```

Comment:
Yeah I think it may end up being further reading there is already a fair amount in week 3. Previously I had mostly focused on uncontrolled components but the consensus was that controlled was the more common pattern so it made sense to focus on that.